storage/store.go

```diff
@@ -180,16 +193,20 @@ func (e *NotBootstrappedError) Error() string {
 }
 
 // storeRangeIterator is an implementation of rangeIterator which
-// cycles through a store's rangesByKey slice.
+// cycles through a store's rangesByKey btree.
 type storeRangeIterator struct {
 	store     *Store
+	stopper   *util.Stopper
 	remaining int
 	index     int
+	ch        chan *Range // Next range is written to this channel.
```

Comment:
Using one unbuffered channel would have the same effect as the pair of channels and simplify things a bit. But I agree that a goroutine and the associated channels and stopper seems clumsy for the iterator use case, especially if we're going to just convert the channel back into a Next()-based interface.